api/generated/v2/types.go

```diff
@@ -376,6 +476,46 @@ type Transaction struct {
 	TxType string `json:"tx-type"`
 }
 
+// TransactionApplication defines model for TransactionApplication.
+type TransactionApplication struct {
+
+	// \[apat\] List of accounts in addition to the sender that may be accessed from the application's approval-program and clear-state-program.
+	Accounts *[]string `json:"accounts,omitempty"`
+
+	// \[apaa\] transaction specific arguments accessed from the application's approval-program and clear-state-program.
+	ApplicationArgs *[]string `json:"application-args,omitempty"`
+
+	// \[apid\] ID of the application being configured or empty if creating.
+	ApplicationId *uint64 `json:"application-id,omitempty"`
+
+	// \[apap\] Logic executed for every application transaction, except when on-completion is set to "clear". It can read and write global state for the application, as well as account-specific local state. Approval programs may reject the transaction.
+	ApprovalProgram *[]byte `json:"approval-program,omitempty"`
```

Comment:
I noticed that some of these generated types have `[]byte`, and others have `*[]byte`. Does this matter?

Reply:
Yeah, the * means it is optional. I forgot to mark required/optional on these.